### PR TITLE
fcosKola: use a unique output directory for each run

### PR DESCRIPTION
**Pull request by dustymabe**
If we try to run mutliple fcosKola() invocations in parallel they
step on each others output directory. Let's write somewhere unique.

### CODE REVIEW

File: vars/fcosKola.groovy [hunk 1]
```diff
@@ -24,6 +24,9 @@ def call(params = [:]) {
         arch = "--arch=${arch}"
     }
 
+    // Create a unique output directory for this run of fcosKola
+    outputDir = shwrapCapture("mktemp -d ${cosaDir}/fcosKola-XXXXX")
```
Comment (dustymabe) [inline vars/fcosKola.groovy:28]: do we need to worry about cleaning this up at the end?

Comment (jlebon) [reply vars/fcosKola.groovy:28]: Maybe

```suggestion
    outputDir = shwrapCapture("mktemp -d ${cosaDir}/tmp/fcosKola-XXXXX")
```

instead?

> do we need to worry about cleaning this up at the end?

I think if we put it in the cosa dir's `tmp/` it should be fine to leave around.

Comment (dustymabe) [reply vars/fcosKola.groovy]: Done

**Review by jlebon — APPROVED** [review]

File: vars/fcosKola.groovy [hunk 2]
```diff
@@ -24,6 +24,9 @@ def call(params = [:]) {
         arch = "--arch=${arch}"
     }
 
+    // Create a unique output directory for this run of fcosKola
+    outputDir = shwrapCapture("mktemp -d ${cosaDir}/tmp/fcosKola-XXXXX")
```
Comment (jlebon) [inline vars/fcosKola.groovy:28]:
```suggestion
    def outputDir = shwrapCapture("mktemp -d ${cosaDir}/tmp/fcosKola-XXXXX")
```

**Review by jlebon — CHANGES_REQUESTED** [review]
Wait, sorry.

File: vars/fcosKola.groovy [hunk 3]
```diff
@@ -52,26 +55,26 @@ def call(params = [:]) {
...
-            shwrap("cd ${cosaDir} && cosa kola run ${rerun} --build=${buildID} ${arch} ${platformArgs} --parallel ${parallel} ${args} ${extraArgs}")
+            shwrap("cd ${cosaDir} && cosa kola run ${rerun} --output-dir=${outputDir} --build=${buildID} ${arch} ${platformArgs} --parallel ${parallel} ${args} ${extraArgs}")
         } finally {
-            shwrap("tar -c -C ${cosaDir}/tmp kola | xz -c9 > ${env.WORKSPACE}/kola.tar.xz")
+            shwrap("tar -c -C ${outputDir} kola | xz -c9 > ${env.WORKSPACE}/kola.tar.xz")
             archiveArtifacts allowEmptyArchive: true, artifacts: 'kola.tar.xz'
         }
         // sanity check kola actually ran and dumped its output in tmp/
-        shwrap("test -d ${cosaDir}/tmp/kola")
+        shwrap("test -d ${outputDir}/kola")
```
Comment (jlebon) [inline vars/fcosKola.groovy:66]: Also, shouldn't this be just

```suggestion
        shwrap("test -d ${outputDir}")
```

now?

And similarly below.

Comment (dustymabe) [reply vars/fcosKola.groovy:66]: not exactly what you suggested, but I reworked it to fix this issue.

**Review by jlebon — APPROVED** [review]

**Pull request merged**